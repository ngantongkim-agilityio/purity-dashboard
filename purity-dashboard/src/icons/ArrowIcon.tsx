import { CustomClassType } from '@/types';

export const ArrowIcon = ({
  customClass = 'w-3 h-3 text-secondary-300'
}: CustomClassType) => (
  <svg fill='none' className={customClass}>
    <path
      stroke='currentColor'
      strokeLinecap='round'
      strokeLinejoin='round'
      strokeWidth={1.125}
      d='M6.281 2.625 9.656 6 6.281 9.375M9.187 6H2.344'
    />
  </svg>
);
