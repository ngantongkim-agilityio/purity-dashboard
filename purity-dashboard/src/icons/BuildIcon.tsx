import { CustomClassType } from '@/types';

export const BuildIcon = ({ customClass = 'w-4 h-4' }: CustomClassType) => (
  <svg fill='none' className={customClass}>
    <g clipPath='url(#a)'>
      <path
        fill='currentColor'
        d='M13.757 4.031a.468.468 0 0 0-.749-.117l-1.8 1.801a.472.472 0 0 1-.663 0l-.778-.779a.469.469 0 0 1 0-.663L11.56 2.48a.468.468 0 0 0-.14-.759c-1.354-.605-3.044-.29-4.11.768-.905.9-1.175 2.304-.74 3.854a.466.466 0 0 1-.13.468l-4.878 4.455a1.88 1.88 0 1 0 2.654 2.655l4.502-4.888A.466.466 0 0 1 9.18 8.9c.442.121.897.184 1.355.187.978 0 1.837-.317 2.456-.927 1.147-1.13 1.322-3.047.766-4.128Zm-10.842 9.57a.937.937 0 1 1-.203-1.864.937.937 0 0 1 .203 1.864Z'
      />
    </g>
    <defs>
      <clipPath id='a'>
        <path fill='#fff' d='M0 .5h15v15H0z' />
      </clipPath>
    </defs>
  </svg>
);
