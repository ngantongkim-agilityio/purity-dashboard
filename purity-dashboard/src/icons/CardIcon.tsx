import { CustomClassType } from '@/types';

export const CardIcon = ({ customClass = 'w-4 h-4' }: CustomClassType) => (
  <svg fill='none' className={customClass}>
    <g fill='currentColor' clipPath='url(#a)'>
      <path d='M.937 11.515a1.64 1.64 0 0 0 1.641 1.64h9.844a1.641 1.641 0 0 0 1.64-1.64V7.004H.937v4.511ZM2.871 9.29a.879.879 0 0 1 .879-.88h1.406a.879.879 0 0 1 .88.88v.586a.879.879 0 0 1-.88.879H3.75a.879.879 0 0 1-.879-.88V9.29ZM12.422 2.843H2.578a1.64 1.64 0 0 0-1.64 1.64v.762h13.124v-.761a1.64 1.64 0 0 0-1.64-1.641Z' />
    </g>
    <defs>
      <clipPath id='a'>
        <path fill='#fff' d='M0 .5h15v15H0z' />
      </clipPath>
    </defs>
  </svg>
);
