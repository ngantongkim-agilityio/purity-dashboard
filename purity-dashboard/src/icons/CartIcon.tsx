import { CustomClassType } from '@/types';

export const CartIcon = ({
  customClass = 'w-6 h-6 text-primary-100'
}: CustomClassType) => (
  <svg fill='none' className={customClass}>
    <path
      fill='currentColor'
      d='M8.484 19.937a1.406 1.406 0 1 0 0-2.812 1.406 1.406 0 0 0 0 2.812ZM18.328 19.937a1.406 1.406 0 1 0 0-2.812 1.406 1.406 0 0 0 0 2.812ZM20.824 5.558a1.052 1.052 0 0 0-.816-.386H6.634l-.27-1.528a.703.703 0 0 0-.692-.581H2.859a.703.703 0 0 0 0 1.406h2.223l2.006 11.372a.703.703 0 0 0 .693.581h11.25a.703.703 0 0 0 0-1.406H8.37l-.248-1.406h10.62a1.058 1.058 0 0 0 1.035-.848l1.266-6.328a1.055 1.055 0 0 0-.22-.876Z'
    />
  </svg>
);
