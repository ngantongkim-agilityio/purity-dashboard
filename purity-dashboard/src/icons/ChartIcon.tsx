import { CustomClassType } from '@/types';

export const ChartIcon = ({
  customClass = 'w-4 h-4 text-primary'
}: CustomClassType) => (
  <svg fill='none' className={customClass}>
    <g fill='currentColor' clipPath='url(#a)'>
      <path d='M3.047 15.032h-.938a.703.703 0 0 1-.703-.704V10.11a.703.703 0 0 1 .703-.703h.938a.703.703 0 0 1 .703.703v4.219a.703.703 0 0 1-.703.703ZM9.61 15.03h-.938a.703.703 0 0 1-.703-.702V7.296a.703.703 0 0 1 .703-.703h.937a.703.703 0 0 1 .703.703v7.032a.703.703 0 0 1-.703.703ZM12.89 15.031h-.937a.703.703 0 0 1-.703-.703V4.015a.703.703 0 0 1 .703-.703h.938a.703.703 0 0 1 .703.704v10.312a.703.703 0 0 1-.703.703ZM6.328 15.031h-.937a.703.703 0 0 1-.704-.703V1.672a.703.703 0 0 1 .704-.703h.937a.703.703 0 0 1 .703.703v12.656a.703.703 0 0 1-.703.703Z' />
    </g>
    <defs>
      <clipPath id='a'>
        <path fill='#fff' d='M0 .5h15v15H0z' />
      </clipPath>
    </defs>
  </svg>
);
