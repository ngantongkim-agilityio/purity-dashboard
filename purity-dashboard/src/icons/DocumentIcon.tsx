import { CustomClassType } from '@/types';

export const DocumentIcon = ({
  customClass = 'w-6 h-6 text-primary-100'
}: CustomClassType) => (
  <svg fill='none' className={customClass}>
    <path
      fill='currentColor'
      stroke='#4FD1C5'
      strokeWidth={0.023}
      d='M13.406 10.106h6.152a.164.164 0 0 1 .164.164v8.262a2.8 2.8 0 0 1-2.8 2.8H7.078a2.8 2.8 0 0 1-2.8-2.8V4.469a2.8 2.8 0 0 1 2.8-2.8h4.043a.164.164 0 0 1 .164.163v6.153a2.121 2.121 0 0 0 2.121 2.12Zm-4.922 7.031h7.031a.715.715 0 0 0 0-1.43h-7.03a.715.715 0 0 0 0 1.43Zm0-3.516h7.031a.715.715 0 0 0 0-1.43h-7.03a.715.715 0 0 0 0 1.43Z'
    />
    <path
      fill='currentColor'
      stroke='#4FD1C5'
      strokeWidth={0.023}
      d='m12.844 2.228 6.32 6.32-6.32-6.32Zm0 0a.076.076 0 0 0-.082-.017.077.077 0 0 0-.048.07v5.705a.691.691 0 0 0 .692.691h5.704a.076.076 0 0 0 .075-.09.076.076 0 0 0-.02-.04l-6.32-6.32Z'
    />
  </svg>
);
