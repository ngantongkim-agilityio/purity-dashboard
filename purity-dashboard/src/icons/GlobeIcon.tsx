import { CustomClassType } from '@/types';

export const GlobeIcon = ({
  customClass = 'w-6 h-6 text-primary-100'
}: CustomClassType) => (
  <svg fill='none' className={customClass}>
    <path
      fill='currentColor'
      stroke='#4FD1C5'
      strokeMiterlimit={10}
      strokeWidth={0.75}
      d='M12 2.359a9.14 9.14 0 1 0 0 18.281A9.14 9.14 0 0 0 12 2.36Z'
    />
    <path
      fill='currentColor'
      stroke='#4FD1C5'
      strokeMiterlimit={10}
      strokeWidth={0.75}
      d='M12 2.359c-2.552 0-4.951 4.092-4.951 9.14 0 5.048 2.399 9.14 4.95 9.14 2.553 0 4.952-4.092 4.952-9.14 0-5.048-2.4-9.14-4.951-9.14Z'
    />
    <path
      fill='currentColor'
      d='M5.906 5.406C7.586 6.599 9.702 7.31 12 7.31c2.298 0 4.413-.711 6.094-1.904'
    />
    <path
      stroke='#4FD1C5'
      strokeLinecap='round'
      strokeLinejoin='round'
      strokeWidth={0.75}
      d='M5.906 5.406C7.586 6.599 9.702 7.31 12 7.31c2.298 0 4.413-.711 6.094-1.904M18.094 17.594c-1.68-1.193-3.796-1.904-6.094-1.904-2.298 0-4.414.711-6.094 1.904'
    />
    <path
      stroke='#4FD1C5'
      strokeMiterlimit={10}
      strokeWidth={0.75}
      d='M12 2.359v18.28M21.14 11.499H2.86'
    />
  </svg>
);
