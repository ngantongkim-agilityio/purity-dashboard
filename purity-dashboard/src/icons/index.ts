export * from './MonIcon';
export * from './SunIcon';
export * from './SearchIcon';
export * from './HomeIcon';
export * from './HelpIcon';
export * from './ChartIcon';
export * from './CardIcon';
export * from './BuildIcon';
export * from './WalletIcon';
export * from './GlobeIcon';
export * from './DocumentIcon';
export * from './CartIcon';
export * from './ArrowIcon';
