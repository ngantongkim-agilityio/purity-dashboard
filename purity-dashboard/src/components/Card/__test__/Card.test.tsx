import { render } from '@testing-library/react';

// Components
import { Card } from '..';
import { CartIcon } from '@/icons';

describe('Card component', () => {
  it('Should render Card component with default props', () => {
    const { container } = render(
      <Card title='Test' value={1} icon={CartIcon} />
    );

    expect(container).toMatchSnapshot();
  });
});
