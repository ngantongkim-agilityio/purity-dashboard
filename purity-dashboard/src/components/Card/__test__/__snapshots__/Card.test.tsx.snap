// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Should render Card component with default props 1`] = `
<div>
  <div
    class="flex justify-between rounded-xl bg-primary-100 p-4 shadow-secondary-50 shadow-md"
  >
    <div>
      <h3
        class="text-xs font-bold text-grey-200"
      >
        Test
      </h3>
      <p
        class="text-lg font-bold text-grey-300"
      >
        1
      </p>
    </div>
    <div
      class="p-2.5 rounded-lg bg-primary"
    >
      <svg
        class="w-6 h-6 text-primary-100"
        fill="none"
      >
        <path
          d="M8.484 19.937a1.406 1.406 0 1 0 0-2.812 1.406 1.406 0 0 0 0 2.812ZM18.328 19.937a1.406 1.406 0 1 0 0-2.812 1.406 1.406 0 0 0 0 2.812ZM20.824 5.558a1.052 1.052 0 0 0-.816-.386H6.634l-.27-1.528a.703.703 0 0 0-.692-.581H2.859a.703.703 0 0 0 0 1.406h2.223l2.006 11.372a.703.703 0 0 0 .693.581h11.25a.703.703 0 0 0 0-1.406H8.37l-.248-1.406h10.62a1.058 1.058 0 0 0 1.035-.848l1.266-6.328a1.055 1.055 0 0 0-.22-.876Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
